core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2536,6 +2536,15 @@ class KafkaApis(val requestChannel: RequestChannel,
         if (exception != null) {
           requestHelper.sendMaybeThrottle(request, consumerGroupHeartbeatRequest.getErrorResponse(exception))
         } else {
+          if (response.assignment != null) {
+            // Remove the unauthorized topics from the assignment.
```

Comment:
nit: `Clients are not allowed to see topics that are not authorized for Describe.`?

Reply:
This case worries me a bit. It means a consumer group member was authorized sufficiently to subscribe and be assigned a partition, but the broker is being a bit shy and doesn't want to admit the existence of the topic to the member. As a result, the topic-partition is assigned to a consumer which is not told about it and as a result will not consume from the topic, and neither will any other member. Do I have this straight?